Mathlib/Data/IsROrC/Basic.lean

```diff
@@ -56,85 +56,80 @@ open ComplexConjugate
 This typeclass captures properties shared by ℝ and ℂ, with an API that closely matches that of ℂ.
 -/
 class IsROrC (K : Type _) extends DenselyNormedField K, StarRing K, NormedAlgebra ℝ K,
-  CompleteSpace K where
+    CompleteSpace K where
   re : K →+ ℝ
   im : K →+ ℝ
-  i : K
-  -- Meant to be set to 0 for K=ℝ
-  i_re_ax : re I = 0
-  i_mul_i_ax : I = 0 ∨ I * I = -1
+  /-- Imaginary unit in `K`. Meant to be set to `0` for `K = ℝ`. -/
+  I : K
+  I_re_ax : re I = 0
+  I_mul_I_ax : I = 0 ∨ I * I = -1
   re_add_im_ax : ∀ z : K, 𝓚 (re z) + 𝓚 (im z) * I = z
-  of_real_re_ax : ∀ r : ℝ, re (𝓚 r) = r
-  of_real_im_ax : ∀ r : ℝ, im (𝓚 r) = 0
+  ofReal_re_ax : ∀ r : ℝ, re (𝓚 r) = r
+  ofReal_im_ax : ∀ r : ℝ, im (𝓚 r) = 0
   mul_re_ax : ∀ z w : K, re (z * w) = re z * re w - im z * im w
   mul_im_ax : ∀ z w : K, im (z * w) = re z * im w + im z * re w
   conj_re_ax : ∀ z : K, re (conj z) = re z
   conj_im_ax : ∀ z : K, im (conj z) = -im z
-  conj_i_ax : conj I = -I
+  conj_I_ax : conj I = -I
   norm_sq_eq_def_ax : ∀ z : K, ‖z‖ ^ 2 = re z * re z + im z * im z
-  mul_im_i_ax : ∀ z : K, im z * im I = im z
+  mul_im_I_ax : ∀ z : K, im z * im I = im z
   inv_def_ax : ∀ z : K, z⁻¹ = conj z * 𝓚 (‖z‖ ^ 2)⁻¹
-  div_i_ax : ∀ z : K, z / I = -(z * I)
+  div_I_ax : ∀ z : K, z / I = -(z * I)
 #align is_R_or_C IsROrC
 
 end
 
-/- failed to parenthesize: unknown constant 'Lean.Meta._root_.Lean.Parser.Command.registerSimpAttr'
-[PrettyPrinter.parenthesize.input] (Lean.Meta._root_.Lean.Parser.Command.registerSimpAttr
-     [(Command.docComment "/--" "Simp attribute for lemmas about `is_R_or_C` -/")]
-     "register_simp_attr"
-     `is_R_or_C_simps)-/-- failed to format: unknown constant 'Lean.Meta._root_.Lean.Parser.Command.registerSimpAttr'
-/-- Simp attribute for lemmas about `is_R_or_C` -/ register_simp_attr is_R_or_C_simps
-
 variable {K E : Type _} [IsROrC K]
 
 namespace IsROrC
 
 open ComplexConjugate
 
+/-- Coercion from `ℝ` to an `IsROrC` field. -/
+@[coe] def ofReal : ℝ → K := algebraMap ℝ K
```

Comment:
What's the benefit of having this vs using `algebraMap` directly as the coercion?

Reply:
I see two reasons:

- without this definition, `simp` will apply lemmas about `algebraMap`; this can change the `simp` behavior to Lean 3;
- I'm not sure that Lean 4 can pretty-print `algebraMap` as an up arrow.

Reply:
> * without this definition, `simp` will apply lemmas about `algebraMap`; this can change the `simp` behavior to Lean 3;

This is a bit complicated, because in Lean 3 `algebraMap` was also a coercion, and because both had the same `coe` head symbol lemmas about one actually did apply to the other.

Reply:
`algebraMap` is a bundled ring homomorphism, so it comes with `map_add`, `map_sum` etc. In Lean 3, it generated a `has_coe` (`has_coe_t`?) instance, so it was possible to use it as a coercion (and this coercion doesn't match with `map_add`/`map_sum`/...).

Reply:
Possibly, I should reuse `Algebra.cast` instead of a new definition.

Reply:
How well does something like `abbrev ofReal := Algebra.cast` work?